torch/csrc/autograd/function.h

```diff
@@ -557,6 +564,38 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
     return false;
   }
 
+#ifdef TORCH_COMPILED_AUTOGRAD
+  // see [Note: Compiled Autograd]
+  // Used by compiled autograd to
+  //   1) Extract tensors/symint args
+  //   2) Collect node information for specialization and caching
+  // Implementations in subclasses should call args.collect() with all node
+  // attrs. These functions are only called durring backward.
+  virtual void compiled_args(CompiledNodeArgs& args) {
+    throw std::runtime_error(
+        std::string("compiled_args not implemented: ") + name());
+  }
+
+  // Used by compiled autograd to call apply() with different saved tensors
+  // Implementations should call saved.before() on all attrs, then apply(), then
```

Comment:
Might be worth mentioning that the order of the before calls must exactly match the collect ones?